pkg/replica/diff_disk.go

```diff
@@ -222,30 +222,42 @@ func (d *diffDisk) fullReadAt(buf []byte, offset int64) (int, error) {
 
 func (d *diffDisk) read(target byte, buf []byte, startOffset int64, startSector int64, sectors int64) (int, error) {
 	bufStart := startSector * d.sectorSize
+
 	bufLength := sectors * d.sectorSize
-	offset := startOffset + bufStart
-	size, err := d.files[target].Size()
+	diskReadStartOffset := startOffset + bufStart
+	diskReadEndOffset := diskReadStartOffset + bufLength
+	dSize, err := d.files[target].Size()
```

Comment:
nit: `dSize` and `vSize` are confusing when reading the codes. Probably can replace them with a better names.